File: aat-android/src/main/java/ch/bailu/aat/views/BusyView.java
package ch.bailu.aat.views;

import android.content.Context;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import ch.bailu.aat.util.ui.AppDensity;

public class BusyView extends LinearLayout {
    private final TextView label;

    public BusyView(Context context) {
        super(context);
        setOrientation(HORIZONTAL);

        AppDensity density = new AppDensity(context);
        int m = density.toPixel_i(2);

        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        lp.setMargins(m,m,m,m);

        ProgressBar progress = new ProgressBar(context, null, android.R.attr.progressBarStyleSmall);
        addView(progress, lp);

        label = new TextView(context);
        addView(label, lp);
    }

    public void setText(CharSequence text) {
        label.setText(text);
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/busy/BusyView.kt
package ch.bailu.aat.views.busy

import android.R
import android.content.Context
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.ProgressBar
import android.widget.TextView
import ch.bailu.aat.util.ui.AndroidAppDensity
import ch.bailu.aat_lib.map.AppDensity

class BusyView(context: Context) : LinearLayout(context) {
    private val label: TextView

    init {
        orientation = HORIZONTAL
        val density: AppDensity = AndroidAppDensity(context)
        val m = density.toPixelInt(2f)
        val lp =
            LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        lp.setMargins(m, m, m, m)
        val progress = ProgressBar(context, null, R.attr.progressBarStyleSmall)
        addView(progress, lp)
        label = TextView(context)
        addView(label, lp)
    }

    fun setText(text: CharSequence) {
        label.text = text
    }
}
